packages/cli/src/build/build.ts

```diff
@@ -55,7 +55,8 @@ export async function build(
 
   const compiledToES5 = (options.js === undefined) ?
       false :
-      options.js.compile === true || options.js.compile === 'es5';
+      options.js.compile === true || options.js.compile === 'es5' ||
+      (options.js.compile && options.js.compile.target === 'es5');
```

Comment:
To compile safely this should be `(typeof options.js.compile === 'object' && options.js.compile.target === 'es5')`.